ee_tests/src/quickstart_deployments.spec.ts

```diff
@@ -40,39 +39,41 @@ describe('Main E2E test suite', () => {
   it('Login', async () => {
     support.info('--- Login ---');
     let login = new support.LoginInteraction();
-    dashboardPage = await login.run();
+    await login.run();
   });
 
   it('Create space ', async () => {
     support.info('--- Create space ' + spaceName + ' ---');
-    spaceDashboardPage = await dashboardPage.createNewSpace(spaceName);
+    await new MainDashboardPage().createNewSpace(spaceName);
   });
 
   it('Create quickstart', async () => {
     support.info('--- Create quickstart ' + quickstart.name + ' ---');
+    let spaceDashboardPage = new SpaceDashboardPage(spaceName);
+    await spaceDashboardPage.open();
+
     let wizard = await spaceDashboardPage.addToSpace();
     await wizard.newQuickstartProject({ project: quickstart.name, strategy });
-    await spaceDashboardPage.ready();
   });
 
   it('Run che', async () => {
     support.info('--- Run che workspace ' + quickstart.name + ' ---');
+    let spaceDashboardPage = new SpaceDashboardPage(spaceName);
+    await spaceDashboardPage.open();
+
     await spaceDashboardPage.codebasesSectionTitle.clickWhenReady();
 
     let spaceChePage = new SpaceChePage();
     await spaceChePage.createCodebase.clickWhenReady(support.LONGEST_WAIT);
 
-    /* A new browser window is opened when Che opens - switch to that new window now */
-    let handles = await browser.getAllWindowHandles();
-    support.debug('Number of browser tabs before = ' + handles.length);
-
-    /* TODO - Need to create a query to look for/wait for the 2nd browser window and remove the sleep statement */
-    await browser.sleep(60000);
-    handles = await browser.getAllWindowHandles();
-    support.debug('Number of browser tabs after = ' + handles.length);
-    support.writeScreenshot('target/screenshots/che_workspace_parta_' + spaceName + '.png');
+    await browser.wait(async () => {
```

Comment:
The core/support file now contains an exported function to count the open windows.

Reply:
Thanks, I've updated the code.